manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -65,9 +65,6 @@ spec:
             allowPrivilegeEscalation: false
             privileged: false
             readOnlyRootFilesystem: true
-            runAsUser: {{ .ProxyUID | default "1337" }}
```

Comment:
I don't think this is equivilent, a lot of policies will deny higher level objects like Deployment early

Reply:
not sure I get it. This template is mixed with the one in the config map, right? So the end result should be the same?

Reply:
The deployment is applied as-is, then the configmap is overlayed once the Pod is created.

But the deployment will never get modified

Reply:
got it. I modified it to use `.Capabilities` to check if this is an OpenShift cluster. Not sure if this is acceptable or not. It worked with `help install`. With `helm template` it will return false, falling back to including the block.

Reply:
It also doesn't work if we use the IstioOperator resource. It only works on `helm install`. This is fine for us.

Reply:
Why don't we use platform=openshift like we do elsewhere?

If its to avoid "global", we should just put it as `.Values.platform.openshift` rather than `.Values.global.platform`

Reply:
@howardjohn done. Hopefully it's fine now.